### PR TITLE
Update Bond to 7.2.0

### PR DESCRIPTION
**Pull request by knutnyg**
## Description
We try again 😃 

## Motivation and Context
Fixes https://github.com/LN-Zap/zap-iOS/issues/73

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Library/Scenes/History/HistoryViewModel.swift [hunk 1]
```diff
@@ -70,10 +70,14 @@ final class HistoryViewModel: NSObject {
         let events = historyService.events
         let filteredEvents = filterEvents(events, searchString: searchString, filterSettings: filterSettings)
         let sectionedCellTypes = bondSections(transactionViewModels: filteredEvents)
-        dataSource.replace(with: Observable2DArray(sectionedCellTypes), performDiff: true)
+        dataSource.replace(with: Array2D<String, HistoryEventType>(sectionedCellTypes), performDiff: true, areValuesEqual: evaluateEqual)
         updateTabBarBadge()
     }
 
+    private func evaluateEqual(one: Array2DElement<String, HistoryEventType>, two: Array2DElement<String, HistoryEventType>) -> Bool {
```
Comment (ottosuess) [inline Library/Scenes/History/HistoryViewModel.swift]: i think it should also be possible to combine both `evaluateEqual` functions into one.
something like this:

```
extension Array2DElement where Section: Equatable, Item: Equatable {
    static func areElementsEqual(one: Array2DElement<Section, Item>, two: Array2DElement<Section, Item>) -> Bool {
```

File: Library/Helper/Bond.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import Bond
+
+public func to2DArraySection<T, U>(section: T, items: [U]) -> TreeNode<Array2DElement<T, U>> {
```
[the remark this answers is not in the document]
Comment (knutnyg) [reply Library/Helper/Bond.swift]: I couldn't get this to work with generics, though my Swift-foo is not very advanced 😇 

What I tried:
```
import Bond
import Foundation

extension TreeNode where Value == Array2DElement<T, U> {

    static func toSection<T, U>(section: T, items: [U]) -> TreeNode<Array2DElement<T, U>> {
        let section = Array2DElement<T, U>.section(section)
        return TreeNode(section, items.map {
            TreeNode(Array2DElement<T, U>.item($0))
        })
    }
    
    init<T,U>(section:T, items:[U]) {
        let _section = Array2DElement<T, U>.section(section)
        TreeNode(_section, items.map {
            TreeNode(Array2DElement<T, U>.item($0))
        })
    }
}
```

[the remark this answers is not in the document]
Comment (ottosuess) [reply Library/Helper/Bond.swift]: i see. my Swift-foo is also not strong enough. let's keep it the way it is.

**Comment by ottosuess** [comment]
🚀

**Pull request merged**